_drafts/the-case-of-nservicebus-long-running-handlers-ocr-processing.md

```diff
@@ -0,0 +1,66 @@
+---
+typora-root-url: ..
+typora-copy-images-to: ../img/posts
+
+layout: post
+title: "The case of NServiceBus long running jobs: OCR Processing."
+author: Mauro Servienti
+synopsis: "Designing systems using a message based architecture is awesome. Messages are a nice way to design human interactions and to model how different components in a domain interact with each other. Unfortunately technology, sometimes, causes more headaches than needed. And when it comes to messaging, long running jobs are a interesting headache to deal with."
+tags:
+- NServiceBus
+- long running handlers
+- sagas
+---
+
+Designing systems using a message based architecture is awesome. Messages are a nice way to design human interactions and to model how different components in a domain interact with each other.
+Unfortunately technology, sometimes, causes more headaches than needed. And when it comes to messaging, long running jobs are a interesting headache to deal with.
+
+### OCR Processing
+
+Users need to process images and extract text using an optical character recognition system (OCR). This is done via a web interface that allows users to upload images for processing.
+
+Images processing takes time and is done in background, when processing is completed users are notified that their background jobs are ready to be consumed.
```

Comment:
Is it the "background job" that is ready to be consumed or the results of that background job? If it's the results then:
> ...when processing is completed users are notified that the results of their background jobs are ready to be consumed.